### PR TITLE
Fix #2718: Remove sync v1.

### LINKED ISSUE #2718: Temporary disable the Sync UI until Sync V2
### Description: 
Temporary disable the Sync UI until Sync V2

@jsecretan can you confirm? WDYT?

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #2718 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

No sync user
- Update from 1.19 to 1.19.1
- Nothing should happen, no alert, sync setting should be gone

Sync user
- Install 1.19 or earlier
- Enable sync chain, can add few bookmarks
- Update to 1.19.1
- Verify that alert message is showing
- Verify that no sync setting is shown, and syncing bookmarks from another device does not work

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by iccub** [comment]
Ready for review, please merge by rebase,

blocking to wait on final decision of what url to use after user taps on 'learn more'

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**
